server/app/src/main/java/com/ams/app/service/StudentService.java

```diff
@@ -0,0 +1,243 @@
+                if(absentdata.contains(attendrecord.getId())){
+                    status = "Absent";
+                }
+                SubjectStatsresponsedto rep = new SubjectStatsresponsedto(attendrecord.getDate(),status,attendrecord.getCreatedon());
+                response.add(rep);
+            }
+        
+            return ResponseEntity.ok(response);
+        }catch(Exception e){
+            System.out.println(e.getMessage());
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,e.getMessage());
+         }
+    }
+
+    public ResponseEntity<List<Studentdashresponsedto>> dashboard(Studentdashrequestdto data){
+         try{
+            List<Studentdashresponsedto> responsedata = new ArrayList<>();
+            Student s = Srepo.findById(data.getSid());
+            List<Attendrecord> records = Arepo.findByDateAndClassname(data.getDate(),s.getClassname());
+            for (Attendrecord attendrecord : records) {
+                 Studentdashresponsedto rep = studentMapper.maptoDashboarddto(attendrecord);
+                 Presentdata p = Prerepo.findByAidAndSid(attendrecord.getId(),s.getId());
+                 if(attendrecord.getStatus() == 0){
+                    if(p != null){
+                        rep.setStudentstatus("Present");
+                    }else{
+                        rep.setStudentstatus("Not yet Marked");
+                    }
+                 }else if(attendrecord.getStatus() == 1) {
+                    if(p != null){
+                        rep.setStudentstatus("Present");
+                    }else{
+                        rep.setStudentstatus("Absent");
+                    }
+                 }else if(attendrecord.getStatus() == 2){
+                    Absentdata a = Abrepo.findByAidAndSid(attendrecord.getId(),s.getId());
+                    if(a == null){
+                        rep.setStudentstatus("Present");
+                    }else{
+                        rep.setStudentstatus("Absent");
+                    }
+                 }
+                responsedata.add(rep);
+            }
+            return ResponseEntity.ok(responsedata);
+         }catch(Exception e){
+            System.out.println(e.getMessage());
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,e.getMessage());
+         }
+    }
+    
+
+    @Autowired
+    private CommonFilterSpecification<Student> StudentFilterSpecification;
+
+    public ResponseEntity<List<Studentdto>> searchStudents(Requestdto request) {
+        try{
+        Specification<Student> searchSpecification = StudentFilterSpecification.
+        getSearchSpecification(request.getSearch(), request.getOperator());
+            
+        return ResponseEntity.ok(studentMapper.modelsToDtos(Srepo.findAll(searchSpecification)));
+        }catch(Exception e){
+            System.out.println(e.getMessage());
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,e.getMessage());
+        }
```

Comment:
We don't need this. This can be achieved by /index method in Controller only

Reply:
Modified